src/metabase/task/QUARTZ.md

```diff
@@ -0,0 +1,153 @@
+             (jobs/with-description "Daily scheduled task")
+             (jobs/store-durably))
+        trigger (triggers/build
+                 (triggers/with-identity daily)
+                 (triggers/for-job daily)
+                 (triggers/start-now)
+                 (triggers/with-schedule
+                  ;; Run every day at 2 AM
+                  (cron/schedule
+                   (cron/cron-schedule "0 0 2 * * ? *")
+                   (cron/in-time-zone (driver/report-timezone)))))]
+    (task/schedule-task! job trigger)))
+```
+
+#### Timezone Considerations
+
+Notice in the example above that we set the timezone explicitly with `(cron/in-time-zone "UTC")`. By default, Quartz schedules are interpreted in the JVM's default timezone, but in Metabase we usually set the timezone to `report-timezone` .
+
+**Important Note:** If your tasks are timezone sensitive, you'll need to rebuild the triggers when `driver/report-timezone` changes. See [[metabase.notification.task.send/update-send-notification-triggers-timezone!]] as an example of how we handle rescheduling.
+
+### 3. Tasks that run once like a migration
+
+We haven't found a good solution for this.
+
+## Advanced Configuration
+
+### Passing Data to Jobs
+
+You can pass data to jobs using the job data map:
+
+```clojure
+(triggers/build
+ ;; other trigger configuration...
+ (triggers/using-job-data {"db-id" database-id}))
+```
+
+Inside the job, you can access this data:
+
+```clojure
+(jobs/defjob MyJob [job-context]
+  (let [data-map (qc/from-job-data job-context)
+        db-id (get data-map "db-id")]
+    ;; Use the data...
+    ))
+```
+
+### Handling Misfires
+
+A misfire occurs when a trigger's scheduled fire time passes without the trigger firing (usually because the scheduler was down). Quartz provides several misfire instructions:
+
+```clojure
+(cron/with-misfire-handling-instruction-do-nothing)        ;; Don't fire missed executions
+(cron/with-misfire-handling-instruction-fire-and-proceed)  ;; Fire once for missed executions
+(cron/with-misfire-handling-instruction-ignore-misfires)   ;; Fire all missed executions
+```
+
+Choose the appropriate misfire handling based on your task's requirements.
+
+### Preventing Concurrent Execution
+
+By default, Quartz allows multiple instances of the same job to run concurrently. To prevent concurrent execution of a job, use the `DisallowConcurrentExecution` annotation:
+
+```clojure
+(jobs/defjob ^{org.quartz.DisallowConcurrentExecution true} MyNonConcurrentJob [job-context]
+  (println "This job will not run concurrently with itself"))
+```
+
+When this annotation is applied, Quartz will block a new instance of the job from starting if a previous instance is still running. This is particularly important for long-running tasks that operate on shared resources.
+
+### Error Handling and Retries
+
+Metabase provides a `rerun-on-error` macro to automatically retry a job if it fails with an exception:
```

Comment:
Worth emphasizing that it's retried *immediately*

Reply:
done